React/Views/RCTBorderDrawing.m

```diff
@@ -506,16 +550,38 @@ static CGContextRef RCTUIGraphicsBeginImageContext(CGSize size, CGColorRef backg
                            CGColorRef backgroundColor,
                            BOOL drawToEdge)
 {
-
-  switch (borderStyle) {
-    case RCTBorderStyleSolid:
-      return RCTGetSolidBorderImage(cornerRadii, viewSize, borderInsets, borderColors, backgroundColor, drawToEdge);
-    case RCTBorderStyleDashed:
-    case RCTBorderStyleDotted:
-      return RCTGetDashedOrDottedBorderImage(borderStyle, cornerRadii, viewSize, borderInsets, borderColors, backgroundColor, drawToEdge);
-    case RCTBorderStyleUnset:
-      break;
+  static NSCache<id, UIImage *> *borderImageCache;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    borderImageCache = [NSCache new];
+  });
+  
+  if (borderStyle == RCTBorderStyleUnset) {
```

Comment:
I would split this function to two: `RCTGetBorderImage` and `RCTGetCachedBorderImage`.